api-put-object-copy.go

```diff
@@ -0,0 +1,76 @@
+/*
+ * Minio Go Library for Amazon S3 Compatible Cloud Storage (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package minio
+
+import (
+	"net/http"
+	"strings"
+)
+
+// CopyObject - copy a source object into a new object with the provided name in the provided bucket
+func (c Client) CopyObject(bucketName string, objectName string, xAmzHeaders http.Header) error {
+	// Input validation.
```

Comment:
> type CopyCondition int16
> const (
> CopyAlways CopyCondition = iota
> CopyIfMatchETag
> CopyIfNoMatchETag
> CopyIfUnmodifiedSince
> CopyIfModifiedSince
> )

The conditions are not true false, they have values. that is why it is a map of key/values. So this approach cannot handle this.